flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/table/AsyncCalcITCase.java

```diff
@@ -238,6 +240,35 @@ public void testFailures() {
         assertThat(results).containsSequence(expectedRows);
     }
 
+    @Test
+    public void testMultiArgumentAsyncWithAdditionalProjection() {
+        // This was the cause of a bug previously where the reference to the sync projection was
+        // getting garbled by janino.
```

Comment:
Could you please link the JIRA issue?

Reply:
Sounds good, added the link

Reply:
Could you please double check you committed the change.

Reply:
My mistake.  I included the link this time.